### PR TITLE
[improve][pip] PIP-313 Support force unsubscribe using consumer api

### PR DESCRIPTION
**Pull request by rdhabalia**
Fixes https://github.com/apache/pulsar/issues/21451

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by liudezhi2098** [comment]
Whether it would be more appropriate to add automatic cleaning subscribers, similar to automatically deleting inactive topics?

**Comment by rdhabalia** [comment]
> Whether it would be more appropriate to add automatic cleaning subscribers, similar to automatically deleting inactive topics?

Inactive topic doesn't have any sub but here in this case, when pulsar spout tries to unsubscribe on a shared sub, unsubscribe request fails first and it will also not close the consumer. so, in this case, both sub and consumer are still live and we should not perform any auto cleanup on such uncertain state.

File: pip/pip_313.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# PIP-313: Support force unsubscribe using consumer api
```
Comment (asafm) [inline pip/pip_313.md:1]: filename should be pip-313.md

**Review by Technoboy- — COMMENTED** [review]
Please update the vote thread link

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**